pkg/config/setup/config.go

```diff
@@ -1818,19 +1816,18 @@ func findUnknownKeys(config pkgconfigmodel.Config) []string {
 	var unknownKeys []string
 	knownKeys := config.GetKnownKeysLowercased()
 	loadedKeys := config.AllKeysLowercased()
-	for _, key := range loadedKeys {
-		if _, found := knownKeys[key]; !found {
-			// Check if any subkey terminated with a '.*' wildcard is marked as known
-			// e.g.: apm_config.* would match all sub-keys of apm_config
-			splitPath := strings.Split(key, ".")
-			for j := range splitPath {
-				subKey := strings.Join(splitPath[:j+1], ".") + ".*"
-				if _, found = knownKeys[subKey]; found {
+	for _, loadedKey := range loadedKeys {
+		if _, found := knownKeys[loadedKey]; !found {
+			nestedValue := false
+			// If a value is within a known key it is considered known.
+			for knownKey := range knownKeys {
+				if strings.HasPrefix(loadedKey, knownKey) {
```

Comment:
Do we want to use this instead?
```suggestion
				if strings.HasPrefix(loadedKey, knownKey + ".") {
```

Otherwise, this will say that "foo" being known implies that "food" is also a known key. But we really want to mean that "foo.bar" is a known key.

Reply:
Good catch ... But then `foo.bar` will not have `foo.bar.` as a prefix :thinking: